setup.py

```diff
@@ -53,6 +53,7 @@ class install_command(install):
                         'none, cuda, cuda-clang, hip'),
         ('vec',   None, 'enable vectorization'),
         ('arch=', None, 'cpu architecture, e.g. haswell, skylake, armv8-a'),
+        ('bundled_libs', None, 'use bundled/git-submoduled 3rd party libraries (ON/OFF)')
```

Comment:
For the pip build, should we use the bundled libs by default? I would go as far as not providing the option to use bundled dependencies.

If not, then we have to add pybind11 and nlohmann/json as dependencies in pip (with all the head aches that brings).

Reply:
Yes we should. Many users (think of the notebooks!) are not going to have these libs installed, and `pip install` not working would make Arbor quickly top my worst-pip-package top1.

As this PR stands, I understand it will use the bundled libs (they have not been removed) and switch them on for a Python package build, so :+1: 

When we get around producing pre-compiled wheels we can consider different options, but until then we cannot require anything for the pip package.

Reply:
If we don't use the bundled versions we would have to make these dependencies that would be installed by pip. I agree that installing Arbor should not fail because a missing dependency isn't available.

The main reason I suggest using the bundled libs is that we would then know that every user who installs version X of Arbor using pip is using version Y of pybind11 etc. This would make dealing with bug reports somewhat easier.

One downside is that if Arbor uses a different version of pybind11 to other modules on the system, there could be strange runtime library issues. But pip isn't great at managing such dependencies, so I don't know how well we could mitigate those efforts even with 'maximum effort".